app/ui/setup.py

```diff
@@ -172,3 +184,208 @@ def sndk_url(account: str, user: str, region: str) -> str:
+    app_name: str, ef_url: str, sf_region: str, sd_deployment: str
+):
+    if config.has_tenant_url():
+        tenant_url = config.get_tenant_url()
+        st.markdown(
+            f"""
+                #### Sundeck account is created
+                To visit Sundeck UI, [right click here]({tenant_url}) and open this link in a new tab/window.
+            """
+        )
+
+    st.markdown(
+        """
+        #### Sundeck-signup using Snowflake-SSO. Follow these steps to enable notifications:
+        1. Prepare to create a Sundeck account. This is needed to for step 2.
+
+        """
+    )
+
+    api_integration_name = "OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS_SSO"
+    if st.button("Enable Sundeck API Integration", key="create_api_integration"):
+        req = perms.request_aws_api_integration(
+            "opscenter_api_integration_sso",
+            (ef_url,),
+            perms.AwsGateway.API_GATEWAY,
+            OPSCENTER_ROLE_ARN,
+            None,
+            api_integration_name,
+            None,
+        )
+        if req is None:
+            print("Successfully created API integration.")
+            st.code(
+                """
+                "Successfully created API integration. Please run the following command in your Snowflake account"
+                """
+            )
+        else:
+            print("Please run the following command in your Snowflake account:")
+            setup_func = "ADMIN.SETUP_REGISTER_TENANT_FUNC()"
+            gateway_prefixes = f"'{ef_url}'"
+            st.code(
+                generate_code_to_setup_external_func(
+                    app_name, gateway_prefixes, api_integration_name, setup_func
+                )
+            )
+
+    st.markdown(
+        """
+        2. To create a free Sundeck account, Please run the following code in your Snowflake account.
+
+        """
+    )
+    st.code(
+        generate_code_to_create_sundeck_account(
+            db=app_name, sf_region=sf_region, sd_deployment=sd_deployment
+        )
+    )
+    st.button("Refresh Status", on_click=config.refresh, key="refresh2")
+
+
+def generate_code_to_create_sundeck_account(
+    db: str, sf_region: str, sd_deployment: str, name: str = "SUNDECK_OAUTH2"
+) -> str:
+    return f"""
+BEGIN  -- Create security integration and Sundeck account
+{generate_security_integration_code(sf_region, sd_deployment, name)}
+{generate_register_tenant_code(db, name)}
+END
+"""
+
+
+def generate_security_integration_code(
+    sf_region: str, sd_deployment: str, name: str
+) -> str:
+    redirect_url = get_redirect_url_for_security_integration(sf_region, sd_deployment)
+    return (
+        f"create security integration if not exists {name} type=oauth enabled=true oauth_client=CUSTOM "
+        + f"oauth_client_type='CONFIDENTIAL' oauth_redirect_uri='{redirect_url}' oauth_issue_refresh_tokens=true "
+        + "oauth_refresh_token_validity=86400 pre_authorized_roles_list = ('PUBLIC'); "
```

Comment:
nit: a big f-string multi-line string here, too? I would also put each attribute to `create security integration` on its own line for readability
```
f"""
...
"""
```

Reply:
Ok, i was only thinking about readability for devs, not presentation to users.